From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: davis <1972806+peoplesmeat@users.noreply.github.com>
Date: Tue, 24 Jan 2023 13:53:23 -0500
Subject: [PATCH] 4928: Remove canonicalHostname for new projects

Remove canonicalHostname for new projects
Addresses https://github.com/NationalSecurityAgency/ghidra/issues/4924

Revert "Remove canonicalHostname for new projects"

Maintain original host in server info

Maintain original host in server info
---
 .../ghidra/framework/client/ClientUtil.java   |  6 ++++--
 .../ghidra/framework/model/ServerInfo.java    | 20 ++++++++++++++++---
 .../framework/data/ProjectFileManager.java    |  2 +-
 .../framework/main/ProjectInfoDialog.java     |  2 +-
 4 files changed, 23 insertions(+), 7 deletions(-)

diff --git a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/client/ClientUtil.java b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/client/ClientUtil.java
index af1056cb4..68d6b7250 100644
--- a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/client/ClientUtil.java
+++ b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/client/ClientUtil.java
@@ -106,9 +106,11 @@ public class ClientUtil {
 		// ensure that default callback is setup if possible
 		getClientAuthenticator();
 
+		String canonicalHost = null;
+
 		host = host.trim().toLowerCase();
 		try {
-			host = InetNameLookup.getCanonicalHostName(host);
+			canonicalHost = InetNameLookup.getCanonicalHostName(host);
 		}
 		catch (UnknownHostException e) {
 			Msg.warn(ClientUtil.class, "Failed to resolve hostname for " + host);
@@ -118,7 +120,7 @@ public class ClientUtil {
 			port = GhidraServerHandle.DEFAULT_PORT;
 		}
 
-		ServerInfo server = new ServerInfo(host, port);
+		ServerInfo server = new ServerInfo(host, canonicalHost, port);
 
 		RepositoryServerAdapter rsa;
 		synchronized (serverHandles) {
diff --git a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/model/ServerInfo.java b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/model/ServerInfo.java
index a912e21e9..a1121c574 100644
--- a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/model/ServerInfo.java
+++ b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/model/ServerInfo.java
@@ -25,25 +25,39 @@ import java.io.Serializable;
 public class ServerInfo implements Serializable {
 
 	private final String host;
+	private final String originalHost;
 	private final int portNumber;
 	
 	/**
 	 * Construct a new ServerInfo object
-	 * @param host host name
+	 * @param originalHost original host name (as entered by a user)
+	 * @param host host name (potentially canonicalized)
 	 * @param portNumber port number
 	 */
-	public ServerInfo(String host, int portNumber) {
+	public ServerInfo(String originalHost, String host, int portNumber) {
+		this.originalHost = originalHost;
 		this.host = host;
 		this.portNumber = portNumber;
 	}
+
+	public ServerInfo(String host, int portNumber) {
+		this(host, host, portNumber);
+	}
 	
 	/**
-	 * Get the server name.
+	 * Get the (canonical) server name.
 	 */
 	public String getServerName() {
 		return host;
 	}
 
+	/**
+	 * Get the (original) server name.
+	 */
+	public String getOriginalServerName() {
+		return this.originalHost;
+	}
+
 	/**
 	 * Get the port number.
 	 */	
diff --git a/Ghidra/Framework/Project/src/main/java/ghidra/framework/data/ProjectFileManager.java b/Ghidra/Framework/Project/src/main/java/ghidra/framework/data/ProjectFileManager.java
index 11c0d51ab..2a3c1e1be 100644
--- a/Ghidra/Framework/Project/src/main/java/ghidra/framework/data/ProjectFileManager.java
+++ b/Ghidra/Framework/Project/src/main/java/ghidra/framework/data/ProjectFileManager.java
@@ -360,7 +360,7 @@ public class ProjectFileManager implements ProjectData {
 		if (info == null) {
 			return;
 		}
-		properties.putString(SERVER_NAME, info.getServerName());
+		properties.putString(SERVER_NAME, info.getOriginalServerName());
 		properties.putString(REPOSITORY_NAME, rep.getName());
 		properties.putInt(PORT_NUMBER, info.getPortNumber());
 		properties.writeState();
diff --git a/Ghidra/Framework/Project/src/main/java/ghidra/framework/main/ProjectInfoDialog.java b/Ghidra/Framework/Project/src/main/java/ghidra/framework/main/ProjectInfoDialog.java
index feacb4ff2..6690859d2 100644
--- a/Ghidra/Framework/Project/src/main/java/ghidra/framework/main/ProjectInfoDialog.java
+++ b/Ghidra/Framework/Project/src/main/java/ghidra/framework/main/ProjectInfoDialog.java
@@ -221,7 +221,7 @@ public class ProjectInfoDialog extends DialogComponentProvider {
 		boolean isConnected = false;
 		if (repository != null) {
 			info = repository.getServerInfo();
-			serverName = info.getServerName();
+			serverName = info.getOriginalServerName();
 			repositoryName = repository.getName();
 			portNumberStr = Integer.toString(info.getPortNumber());
 			isConnected = repository.isConnected();
-- 
2.39.1

